### PR TITLE
Made GCN/GAT more consistent with original

### PR DESCRIPTION
**Pull request by jackd**
This contains 3 main components:
- added `dtype` arg to `Citations` datasets with `np.float32` default, affecting all graphs. This should speed up tf training since most examples were training in `tf.float64`, and remove the `tf 2.1` requirement to specify dtypes to keras inputs (since the default data provided will be the same `dtype` as tensorflow's default)
- added `normalize_x=True` to `GAT` / `GCN` examples
- fixed minor inconsistencies in `GAT` examples, added weighting / seeds.

I could have done those all in separate PRs, but I'm trying to find a balance between single-issue PRs and avoiding merge conflicts. Let me know if I've got the balance wrong :).

Results for GCN / GAT below. Note there are some minor differences between the original GAT implementation and the one in this repository as discussed [here](https://github.com/danielegrattarola/spektral/issues/146).

| Impl. 	| Dataset  	| Reported 	| Ours 	|
|-------	|----------	|----------	|------	|
| GCN   	| Citeseer 	| 70.3     	| 71.4 	|
| GCN   	| Cora     	| 81.5     	| 82.3 	|
| GCN   	| Pubmed   	| 79.0     	| 79.3 	|
| GAT   	| Citeseer 	| 72.5     	| 70.8 	|
| GAT   	| Cora     	| 83.0     	| 82.8 	|
| GAT   	| Pubmed   	| 78.7     	| 79.0 	|

### CODE REVIEW

**Comment by danielegrattarola** [comment]
>Let me know if I've got the balance wrong :)

Looks good to me!

Thanks for taking the time to go through the examples. The results are very good, despite the differences in #146. We will eventually report these values in the benchmark directory discussed in #148, I expect that the differences w.r.t. the original values are not statistically significant. 

Cheers
